components/ProgressView/src/MDCProgressView.m

```diff
@@ -290,4 +420,31 @@ - (void)updateTrackView {
   self.trackView.frame = self.hidden ? CGRectMake(0.0, size.height, size.width, 0.0) : self.bounds;
 }
 
+- (void)startAnimatingBar {
+  // If the bar isn't indeterminate or the bar is already animating, don't add the animation again.
+  if (_mode == MDCProgressViewModeDeterminate || _isAnimating) {
+    return;
+  }
+
+  float barWidth = [self indeterminateLoadingBarWidth];
+  CGRect progressBarStartFrame = CGRectMake(-barWidth, 0, barWidth, CGRectGetHeight(self.bounds));
+  self.progressView.frame = progressBarStartFrame;
+
+  CGPoint progressBarEndPoint =
+      CGPointMake(self.progressView.layer.position.x + CGRectGetWidth(self.bounds) + barWidth,
+                  CGRectGetHeight(self.bounds) / 2);
+  CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:@"position"];
+  animation.fromValue = [NSValue valueWithCGPoint:self.progressView.layer.position];
+  animation.toValue = [NSValue valueWithCGPoint:progressBarEndPoint];
+  animation.duration = kAnimationDuration;
+  animation.repeatCount = HUGE_VALF;
+  [self.progressView.layer addAnimation:animation forKey:@"position"];
+  _indeterminateAnimationStartTime = [self.progressView.layer convertTime:CACurrentMediaTime()
+                                                                fromLayer:nil];
+}
+
+- (float)indeterminateLoadingBarWidth {
+  return CGRectGetWidth(self.bounds) * .52;
```

Comment:
Please move this constant up to a `static const` at the top of the file and document it.